src/vorta/utils.py

```diff
@@ -240,7 +241,25 @@ def sort_sizes(size_list):
     return final_list
 
 
-def pretty_bytes(size, metric=True, sign=False, precision=1):
+def find_best_size_unit(sizes: Iterable[int], metric: bool = True, precision: int = 1) -> int:
+    """
+    Selects the index of the biggest unit (see the lists in the pretty_bytes function) capable of
+    representing the smallest size in the sizes iterable.
+    """
+    power = 10**3 if metric else 2**10
+    min_size = min((s for s in sizes if isinstance(s, int)), default=None)
+    if min_size is None:
+        return 0
+    n = math.floor(math.log(min_size * 10**precision, power))
+    return n
```

Comment:
n might be of metric range, i think

Reply:
not sure what you mean... those lines should work both for metric and power of 2, do you think I'm missing some test cases?

Reply:
Sorry left out some words, updated the comment.

Reply:
ok, yes i resolved by clamping in the pretty_bytes function